### PR TITLE
fix: Avoid race condition on filename validation

### PR DESCRIPTION
**Pull request by doubleface**
When pressing enter key on filename creation we would get an unwanted notification for empty filename even if not empty.

I simplified the handling of these events and now :
- we save the file when pressing enter and the file name is not empty
- we display the "no empty file name" notification when pressing enter with an empty file name
- we save the file when bluring the component and the file name is not empty
- we just blur without notification on blur with empty file name

https://www.notion.so/linagora/Toast-when-creating-folder-29562718bad18055a7ccfcf0c523489b

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved filename input behavior: Enter, Escape and blur now consistently submit or abort based on the field value (whitespace is treated as content). Added a submission guard to prevent duplicate submissions and race conditions; submitting state is reliably cleared on completion or error.

* **Tests**
  * Added comprehensive tests covering Enter, Escape, blur, typing, whitespace edge cases and race-condition scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/modules/filelist/FilenameInput.jsx [hunk 1]
```diff
@@ -35,11 +34,13 @@ class FilenameInput extends Component {
...
 
-    if (e.keyCode === ENTER_KEY && !valueIsEmpty(value)) {
-      this.setState({ hasBeenSubmitedOrAborted: true })
-      this.submit()
+    if (e.keyCode === ENTER_KEY) {
+      if (valueIsEmpty(value)) {
+        this.abort(true)
+      } else {
+        this.submit()
+      }
```
[the remark this answers is not in the document]
Comment (rezk2ll) [reply src/modules/filelist/FilenameInput.jsx:43]: this won't work since React batches the state updates and executes the event handlers an then update the state.

in the case of handleBlur the hasSubmittedOrAborted will always have the previous value and not the value updated by the handleKeyDown event handler

[the remark this answers is not in the document]
Comment (rezk2ll) [reply src/modules/filelist/FilenameInput.jsx:43]: how about useRef ? :eyes:

[the remark this answers is not in the document]
Comment (rezk2ll) [reply src/modules/filelist/FilenameInput.jsx:43]: we will do that in a different PR :eyes:

File: src/modules/filelist/FilenameInput.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      const input = screen.getByRole('textbox')
+
+      // Type whitespace and press Enter
+      fireEvent.change(input, { target: { value: '   ' } })
+      fireEvent.keyDown(input, { keyCode: 13 })
+
+      // Whitespace is considered non-empty by the component
+      expect(onSubmit).toHaveBeenCalledWith('   ')
+    })
+  })
```
[the remark this answers is not in the document]
Comment (Coussecousse) [reply src/modules/filelist/FilenameInput.spec.jsx:167]: write me a fiction about this pr

**Review by rezk2ll — APPROVED** [review]
